src/main/java/edu/umich/its/cpm/MigrationTaskService.java

```diff
@@ -1297,61 +1309,122 @@ private JSONObject getMailArchiveZipContent(Environment env, String site_id,
+						String mboxMessage = (String) MboxFormatTextPlusStatus.get(0);
+						JSONObject singleMboxMsgStatus = (JSONObject) MboxFormatTextPlusStatus.get(1);
+						singleMboxMsgStatus.put(Utils.JSON_ATTR_ITEM_ID, messageId);
+						if(mboxMessage!=null) {
+							msgBundle.append(mboxMessage);
+							msgBundle.append("\r\n");
+							mboxMessagesStatus.put(singleMboxMsgStatus);
+						}else {
+							mboxMessagesStatus.put(singleMboxMsgStatus);
+							log.error(String.format("Mbox Formatting for message with Id (%s) not successful with migrationid " +
+									"(%s) for site (%s)",messageId,migrationId,site_id));
+						}
 					}
+					mboxMessagesStatus = handleMailArchiveMboxMessage(out, msgBundle.toString(), messageFolderName,
+								mboxMessagesStatus);
 
-					messageStatus = handleMailArchiveMboxMessage(out, msgBundle.toString(), messageFolderName,
-							messageStatus);
-					messagesStatus.put(messageStatus);
+					downloadStatus= finalreportObjBuilderForMailZipMigration(downloadStatus, mboxMessagesStatus);
 
 				}
 			}
-
-			downloadStatus.put(Utils.MIGRATION_DATA, messagesStatus);
-			downloadStatus.put(Utils.MIGRATION_STATUS, messagesStatus.toString().contains(Utils.STATUS_FAILURE)? Utils.STATUS_FAILURE: Utils.STATUS_SUCCESS);
-
 			return downloadStatus;
 		}
 
-		/**
+	private JSONObject finalreportObjBuilderForMailZipMigration(JSONObject downloadStatus, JSONArray mboxMessagesStatus) {
+		JSONArray errAndPartialSuccessList = new JSONArray();
+		int successes,errors,partial;
+		successes = errors=partial=0;
+		for (int i = 0; i < mboxMessagesStatus.length(); i++) {
+            JSONObject perMsg = mboxMessagesStatus.getJSONObject(i);
+            String msgStatus = (String) perMsg.get(Utils.JSON_ATTR_ITEM_STATUS);
+            if (msgStatus.equals(Utils.STATUS_OK)) {
+                successes = successes + 1;
+            } else if (msgStatus.equals(Utils.STATUS_PARTIAL)) {
+                partial = partial + 1;
+                errAndPartialSuccessList.put(mboxMessagesStatus.get(i));
+            } else if (msgStatus.equals(Utils.STATUS_ERROR)) {
+                errors = errors + 1;
+                errAndPartialSuccessList.put(mboxMessagesStatus.get(i));
+            }
+        }
+		JSONObject counts = new JSONObject();
+		counts.put(Utils.STATUS_SUCCESSES,successes);
+		counts.put(Utils.STATUS_ERRORS,errors);
+		counts.put(Utils.STATUS_PARTIALS,partial);
+		downloadStatus.put(Utils.JSON_ATTR_COUNTS,counts);
+		downloadStatus.put(Utils.JSON_ATTR_ITEMS,errAndPartialSuccessList);
+		if (errors > 0) {
+            downloadStatus.put(Utils.MIGRATION_STATUS, Utils.STATUS_ERROR);
+        } else if (partial > 0 & successes > 0) {
+            downloadStatus.put(Utils.MIGRATION_STATUS, Utils.STATUS_PARTIAL);
+        } else
+            downloadStatus.put(Utils.MIGRATION_STATUS, Utils.STATUS_OK);
+		return downloadStatus;
+	}
+
+	private String getDate(JSONObject message) {
```

Comment:
These are the same method if you pass in the value of the JSON_ATTR you are retrieving.